functions/src/api/v1/rooms/joinRoom.ts

```diff
@@ -51,19 +54,38 @@ export interface JoinRoomResponse {
 export const joinRoom = wrap(async (req: Request, res: Response) => {
     const roomPassword = req.body.password
     const participantUID = res.locals.uid
+    const roomId = req.params.roomId
     if (!roomPassword) {
         throw new ForbiddenError('Unable to join a room without a password')
     }
-    const room = await getRoom(req.params.roomId)
+
+    const room = await getOrCreateRoom(roomId, participantUID, roomPassword)
 
     if (room.password !== roomPassword) {
         throw new ForbiddenError('Wrong password to join the room')
     }
 
-    const accessToken = createTwilioClientToken(participantUID, room)
+    const accessToken = createTwilioClientToken(participantUID, room.id)
 
     res.send({
         jwtAccessToken: accessToken.toJwt(),
         ttl: TTL_ACCESS_TOKEN_PARTICIPANT_SECONDS,
-    } as JoinRoomResponse)
+    })
 })
+
+const getOrCreateRoom = async (
+    roomId: RoomId,
+    participantUID: UID,
+    roomPassword: string
+): Promise<Room> => {
+    try {
+        return await getRoom(roomId)
```

Comment:
no need to `return await` in an async function, `return` only is enough

Reply:
I added a rule [here](https://github.com/Instant-Visio/InstantVisio-WebApp/pull/306) not to forget to fix this one